### PR TITLE
Testgen -- add abstract time and various other updates

### PR DESCRIPTION
**Pull request by andrey-kuprianov**
Part 3 of #529. See also #414. Various changes to Testgen, most notably make the time abstract, in order to correctly form time from the Apalache counterexamples. 

* [X] Referenced an issue explaining the need for the change
* [X] Updated all relevant documentation in docs
* [X] Updated all code comments where relevant
* [ ] Wrote tests
* [ ] Updated CHANGELOG.md

### CODE REVIEW

File: testgen/src/vote.rs [hunk 1]
```diff
@@ -86,10 +85,21 @@ impl Generator<vote::Vote> for Vote {
...
         let validator_index = match self.index {
             Some(i) => i,
-            None => match header.validators.as_ref().unwrap().iter().position(|v| *v == *validator) {
+            None => match header
+                .validators
+                .as_ref()
+                .unwrap()
+                .iter()
+                .position(|v| *v == *validator)
+            {
```
Comment (romac) [inline testgen/src/vote.rs:94]: Can we extract this long chained call into its own variable for better readability?

File: testgen/src/commit.rs [hunk 2]
```diff
@@ -119,9 +128,19 @@ impl Generator<block::Commit> for Commit {
                 signature: vote.signature,
             })
         };
-        let sigs = votes
+        let val_to_sig = |val: &Validator| -> Result<block::CommitSig, SimpleError> {
+            if let Some(vote) = votes
+                .iter()
+                .find(|&vote| vote.validator.as_ref().unwrap() == val)
```
Comment (romac) [inline testgen/src/commit.rs:134]: Can we extract this chained call into a variable for better readability?

**Review by romac — CHANGES_REQUESTED** [review]
Looks great! Just a couple nitpicks, on top of Clippy's :)

File: testgen/src/vote.rs (hunk 1 above)
Comment (andrey-kuprianov) [reply testgen/src/vote.rs:94]: thanks:) done

File: testgen/src/commit.rs (hunk 2 above)
Comment (andrey-kuprianov) [reply testgen/src/commit.rs:134]: done

**Comment by andrey-kuprianov** [comment]
@romac I've addressed your comments; could you please take a look and merge? I will update the changelog when the last PR of the MBT series (546-551) is closed. Thanks!

**Review by romac — APPROVED** [review]

**Pull request merged**